corehq/apps/importer/util.py

```diff
@@ -95,6 +97,25 @@ def from_request(cls, request):
 
         )
 
+
+def open_workbook(filename=None, **kwargs):
+    _soft_assert = soft_assert(notify_admins=True)
+    try:
+        return xlrd.open_workbook(filename=filename, **kwargs)
+    except xlrd.XLRDError as e:
+        message = unicode(e)
+        if message == u'Workbook is encrypted':
+            raise ImporterExcelFileEncrypted(message)
+        else:
+            _soft_assert(False, 'displaying XLRDError directly to user', e)
+            raise ImporterExcelError(message)
```

Comment:
Nope, this will not always work:
```py
except xlrd.XLRDError as e:
    if ...
    else:
        raise
```
Python 2 does a silly thing. Try out this code and see what you get:
```py
try:
    raise Exception("blow up")
except:
    try:
        raise Exception("should not be re-raised")
    except:
        pass
    raise
```
Python 3 fixes that issue.